frontend/public/module/k8s/k8s-models.ts

```diff
@@ -57,27 +59,53 @@ const getK8sModels = () => {
  * NOTE: This will not work for CRDs defined at runtime, use `connectToModels` instead.
  */
 export const modelFor = (ref: K8sResourceKindReference) => {
+  const dynamicModelMetadata = pluginStore
+    .getAllExtensions()
+    .filter(isModelMetadata)
+    ?.find(
+      (e) =>
+        ref === referenceForExtensionModel(e?.properties?.model) ||
+        ref === e?.properties?.model?.kind,
+    )?.properties;
```

Comment:
It's possible that there may be multiple matches.
eg. An extension could contribute the tech preview badge to all resources with a given api group. A second extension could then provide specific labels for the same resource.

Reply:
how should we deal with this then?

Reply:
You'll need to combine all valid matches of metadata together.